datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -796,4 +808,38 @@ mod tests {
             output_ordering: None,
         }
     }
+
+    #[test]
+    fn file_groups_display_empty() {
+        let expected = "";
+        assert_eq!(&FileGroupsDisplay(&vec![]).to_string(), expected);
+    }
+
+    #[test]
+    fn file_groups_display() {
+        let files = vec![
+            vec![partitioned_file("foo"), partitioned_file("bar")],
+            vec![partitioned_file("baz")],
+            vec![],
+        ];
+
+        let expected = "group 0: [foo, bar], group 1: [baz], group 2: []";
```

Comment:
Prior to the changes in this PR, this was displayed as `"[foo, bar, baz, ]"`


I think this PR is much clearer